src/openrct2/object.h

```diff
@@ -65,7 +65,10 @@ typedef enum
  * size: 0x10
  */
 typedef struct rct_object_entry {
-    uint32 flags;
+    union {
+        uint8 end_flag; // needed not to read past allocated buffer.
```

Comment:
But it does. the `flags` field is 4 bytes wide and generates an appropriate read/write, but you only allocate space for one more byte when saving or loading, so you end up walking past available memory. With this code you only read single byte, which stays within bounds. Indeed, `& 0xFF` is now superfluous, so I will remove it.